app/src/main/scala/org/alephium/explorer/service/TransactionHistoryService.scala

```diff
@@ -166,79 +167,95 @@ case object TransactionHistoryService extends StrictLogging {
   }
 
   private def processDataForTimeRanges(
-      data: Seq[(TimeStamp, Int, Int, Long)],
+      data: Seq[(TimeStamp, Int, Int, Long, Option[Long])],
       ranges: Seq[(TimeStamp, TimeStamp)],
       intervalType: IntervalType
-  )(implicit gs: GroupSetting): Seq[(TimeStamp, Int, Int, Long, IntervalType)] = {
+  )(implicit gs: GroupSetting): Seq[(TimeStamp, Int, Int, Long, Long, IntervalType)] = {
     ranges.flatMap { case (from, to) =>
       val perChainCount = gs.chainIndexes.map { chainIndex =>
+        val (count, nonCoinbaseCount) = data
+          .filter { case (timestamp, chainFrom, chainTo, _, _) =>
+            timestamp >= from &&
+            timestamp <= to &&
+            chainFrom == chainIndex.from.value &&
+            chainTo == chainIndex.to.value
+          }
+          .foldLeft((0L, 0L)) {
+            case ((currentCount, currentNonCoinbaseCount), (_, _, _, count, nonCoinbaseCount)) =>
+              (currentCount + count, currentNonCoinbaseCount + nonCoinbaseCount.getOrElse(0L))
+          }
+
         (
           chainIndex.from.value,
           chainIndex.to.value,
-          data
-            .filter { case (timestamp, chainFrom, chainTo, _) =>
-              timestamp >= from &&
-              timestamp <= to &&
-              chainFrom == chainIndex.from.value &&
-              chainTo == chainIndex.to.value
-            }
-            .map { case (_, _, _, count) => count }
-            .sum
+          count,
+          nonCoinbaseCount
         )
       }
-      val allCount = (-1, -1, perChainCount.map { case (_, _, count) => count }.sum)
 
-      (perChainCount :+ allCount).map { case (chainFrom, chainTo, count) =>
-        (from, chainFrom, chainTo, count, intervalType)
+      val (count, nonCoinbaseCount) = perChainCount.foldLeft((0L, 0L)) {
+        case ((currentCount, currentNonCoinbaseCount), (_, _, count, nonCoinbaseCount)) =>
+          (currentCount + count, currentNonCoinbaseCount + nonCoinbaseCount)
+      }
+
+      val allCount = (-1, -1, count, nonCoinbaseCount)
+
+      (perChainCount :+ allCount).map { case (chainFrom, chainTo, count, nonCoinbaseCount) =>
+        (from, chainFrom, chainTo, count, nonCoinbaseCount, intervalType)
       }
     }
   }
 
   private def getTxsCountFromTo(
       from: TimeStamp,
       to: TimeStamp
-  ): DBActionSR[(TimeStamp, Int, Int, Long)] = {
+  )(implicit ec: ExecutionContext): DBActionSR[(TimeStamp, Int, Int, Long, Option[Long])] = {
     sql"""
       SELECT block_timestamp, chain_from, chain_to, txs_count
       FROM block_headers
       WHERE block_timestamp >= $from
       AND block_timestamp <= $to
       AND main_chain = true
-    """.asAS[(TimeStamp, Int, Int, Long)]
+    """
+      .asAS[(TimeStamp, Int, Int, Long)]
+      .map(_.map { case (ts, from, to, value) =>
+        (ts, from, to, value, Some(value - 1))
```

Comment:
That's were we get the `txsCount` for each block, as a block as only 1 coinbase tx, we can simply remove 1.